zbar/zbar.py

```diff
@@ -171,7 +171,7 @@ def scan(self, image):
             sym_name = str(_ZB.zbar_get_symbol_name(sym_type), 'ascii')
             sym_data_ptr = _ZB.zbar_symbol_get_data(symbol)
             sym_data_len = _ZB.zbar_symbol_get_data_length(symbol)
-            sym_data = ctypes.string_at(sym_data_ptr, sym_data_len)
+            sym_data = ctypes.string_at(sym_data_ptr, sym_data_len).decode("utf-8")
```

Comment:
There's no guarantee that an arbitrary barcode's data will be text encoded as UTF-8. It could be raw binary data, or text in some other encoding. So if a client of zbar knows the data from their barcode is UTF-8 encoded, they should call decode, but this library shouldn't be making that assumption.

Indeed, the docstring for `scan()` specifically states: "data' is a bytes instance containing the barcode payload", so this counts as a non-backward-compatible API change.

For those reasons, I think that this change should be reconsidered.